src/main/java/org/terasology/herbalism/system/HerbDropAuthoritySystem.java

```diff
@@ -166,12 +153,13 @@ public void onGeneratedHerbDestroyed(DoDestroyEvent event, EntityRef entity, Gen
         if (random.nextFloat() < chanceOfBlockDrop) {
             // Get the base genome and world position of the old herb.
             final String herbBaseGenome = herbComp.herbBaseGenome;
-            final Vector3f position = locationComp.getWorldPosition();
+            final Vector3f position = locationComp.getWorldPosition(new Vector3f());
 
             // Using a BiodiversityGenerator, create a new set of (mutated) genes.
             BiodiversityGenerator generator = new BiodiversityGenerator(worldProvider.getSeed(), 0, new HerbGeneMutator(), herbBaseGenome,
                     3, 0.0002f);
-            final String generatedGenes = generator.generateGenes(new Vector2i(TeraMath.floorToInt(position.x + 0.5f), TeraMath.floorToInt(position.z + 0.5f)));
+            final String generatedGenes = generator.generateGenes(new Vector2i(new Vector2f(position.x + 0.5f, position.z + 0.5f),
+                RoundingMode.FLOOR));
```

Comment:
This would then become
```suggestion
            final String generatedGenes = generator.generateGenes(new Vector2i(position.x, position.y));
```